gh-141004: Document `PyType_Unwatch` (#141414)
python/cpython: Doc/c-api/type.rst

   .. versionadded:: 3.12

.. c:function:: int PyType_Unwatch(int watcher_id, PyObject *type)

   Mark *type* as not watched. This undoes a previous call to

>    :c:func:`PyType_Watch`. *type* must not be ``NULL``.

Reviewer: ```suggestion
   :c:func:`PyType_Watch`. *type* must be a type object.
```
IMO this is more exact, since we check `if (!PyType_Check(obj)) { complain }`.

Author: That reads to me as "if `type` is not a type object, this will crash", which is not correct. I like it as-is.
